base/applications/cleanmgr/main.c

```diff
@@ -36,54 +36,64 @@ int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance,
         return FALSE;
     }
 
-    if (Obj)
+    /* Creating a mutex of this specific program to check if multiple instances are running */
+    CreateMutexW(NULL, FALSE, L"cleanmgr.exe");
+    if (hMutex)
     {
+        /* If there is another instance of the program running, then just broadcast the 
+           registered window message to all of the valid dialog boxes. */
         if (GetLastError() == ERROR_ALREADY_EXISTS)
         {
             DPRINT("An instance already exists!\n");
             SendMessageW(HWND_BROADCAST, CleanmgrWindowMsg, 0, 0);
-            CloseHandle(Obj);
+            CloseHandle(hMutex);
             return TRUE;
         }
     }
     else
     {
-        CloseHandle(Obj);
+        CloseHandle(hMutex);
     }
 
+    /* Gather the program arguments which has been provided by the user */
     ArgList = CommandLineToArgvW(GetCommandLineW(), &nArgs);
-
     if (ArgList == NULL)
     {
         return FALSE;
     }
     else if (nArgs > 1)
     {
-        if(!UseAquiredArguments(ArgList, nArgs))
+        if (!UseAcquiredArguments(ArgList, nArgs))
         {
-            return FALSE;
+            goto START;
```

Comment:
the goal of refactoring code is cleaning up, not making a bigger mess....
Why would you start with 'goto' all of a sudden?

Reply:
@learn-more That's for starting cleanmgr in the normal way if the user has provided invalid argument,